src/reducers/lobby.js

```diff
@@ -47,13 +48,29 @@ export default function lobbyReducer(state = initialState(), action) {
 
   case START_NEW_GAME:
     const teams = [RED, BLUE];
-    // make sure we have the right sorts of players
+    let numSpymasters = 0;
     teams.forEach(team => {
+      // make sure we have the right sorts of players
       const members = ofTeam(state.players, team);
-      if (spymastersOf(members).length !== 1) throw new WrongNumberOfSpymasters(team);
+      if (spymastersOf(members).length !== 1) numSpymasters += 1;
       if (guessersOf(members).length < 1) throw new WrongNumberOfGuessers(team);
     });
-    // ok seems legit
+
+    if (numSpymasters === 0) {
+      // if there are no spymasters chosen, shuffle teams and spymasters
+      let players = shuffled(state.players);
+      state.players = [];
```

Comment:
this is redux. we don't do mutations in redux. `state` should be considered immutable